docs/_articles/en/einstein/einstein-example-prompts.md

```diff
@@ -1,23 +1,121 @@
 
+## Project Context and Grounding
+
+In addition to your instructions, the response generated by an AI depends on the context that your instruction is grounded with. The CodeGen2.5 model has been trained on Apex data and grounded with the context of Apex programming best practices.
+
+To extract accurate and relevant information from the model, first run **SFDX:Refresh SObjects Definitions** to gather org metadata in the form of object definitions into your project before running your query. Then use the following tips to help ground your prompts further with org context awareness to generate a high-quality response.
+
+Currently the CodeGen2.5 model can pick up on the following context from your project and prompt:
+
+- Apex code that’s present in your project in Apex class files (`.cls`).
+- Custom objects in the schema that are mentioned in the `object__c` format. Schema information takes precedence over Apex code in the current file.
+- Custom metadata when it is a part of a prompt and uses the `__mdt` suffix. The metadata added to a prompt in this way is automatically included into the context to ground generations.
+
 ## Prompt Writing Tips
 
-Prompt writing is key to quality code generation. Here are some tips for improved prompt writing and a selection of sample prompts to start you on your generative AI code-assistance journey.
+We’re listing some tips for prompt writing and also providing you with some examples that help you craft your prompts. An effective prompt consists of these four parts:
+
+1. **The Action**: The task you want the AI to accomplish.
+
+For example:
+_Create an Apex batch class named "OpenPositionMatcher"_.
+
+2. **Context Details**: Information about the context you want the AI to use. This includes important context such as org metadata and awareness about the presence of custom objects and fields in your org. Run **SFDX:Refresh SObjects Definitions** to retrieve org metadata into your project. You can then add custom fields to your existing query.
+
+For example:
+_Create an Apex batch class named "OpenPositionMatcher" that processes "Position\_\_c" records created within the past 15 days. The class should match these open positions with suitable candidates based on their skills._
+
+3. **Best Practices**: Details about best practices you want the AI to follow. For example, security and formatting best practices and so on.
+
+For example:
+_Create an Apex batch class named "OpenPositionMatcher" that processes custom objects "Position\_\_c" records created within the past 15 days. The class aims to match these open positions with suitable candidates based on their skills. Order the results by candidate names in descending order. Make sure the generated code follows security best practices. Separate operators with a new line._
```

Comment:
One thing that's kind of complex about this is we're altering the users prompt to be a more general LLM prompt.  From what I understand the final thing we ask the LLM to do is the action.  I don't know for sure, but logically it would make sense to include all direction first, then end with the final action to be performed.  So I might reorder the above to be something like

Results should be ordered by candidate names in descending order.  Ensure the generated code follows security best practices. Separate operators with a new line.  Create an Apex batch class named "OpenPositionMatcher" that processes custom objects "Position\_\_c" records created within the past 15 days. The class aims to match these open positions with suitable candidates based on their skills. 

This might be crazy talk :)